### DISCUSSION TITLE
How to skip further form validation if CSRF token is invalid

### DISCUSSION THREAD

**Discussion by va5ja** [symfony/symfony#48869] [Q&A]
### Use case
I would like to skip further form validation if CSRF token fails. There's no need to validate any further if the user has no valid token.

### Background
A form can have CSRF validation. `FormTypeCsrfExtension` adds `CsrfValidationListener` subscriber which adds the CSRF field to the form and also performs the token validation. On `PRE_SUBMIT` event the CSRF token is checked and if validation fails an error is already added to the form. So a form can already have errors before other fields are validated.

### Options from documentation
* https://symfony.com/doc/current/form/disabling_validation.html It is possible to disable validation.
* https://symfony.com/doc/current/form/data_based_validation.html It is possible to choose validation groups based on the submitted data.

### What I've tried
I tried to do the following:

`config/services.yaml`
```
App\Form\Extension\FormTypeExtension:
      tags:
        - { name: 'form.type_extension', priority: -100 }
```

`src/Form/Extension/FormTypeExtension.php`
```
...
    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'validation_groups' => function (FormInterface $form) {
                return $form->isRoot() && $form->getErrors()->count() ?
                    [] :
                    $form->getConfig()->getOption('validation_groups');
            },
        ]);
    }
...
```
Unfortunately this results in an unexpected behavior. In some cases an error is thrown:
`TypeError: "Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateInGroup(): Argument #4 ($group) must be of type string, Closure given`

Has anyone an idea how this can be achieved?

**Comment by MatTheCat** [comment]
Not sure it is possible, but it is generally considered good UX to display all errors at once (see https://design-system.service.gov.uk/patterns/validation/#how-to-tell-the-user-about-validation-errors eg).

Do you have a particular use-case going against this pattern?
